qiskit/algorithms/gradients/finite_diff_sampler_gradient.py

```diff
@@ -0,0 +1,104 @@
+
+    def __init__(
+        self,
+        sampler: BaseSampler,
+        epsilon: float = 1e-2,
+        **run_options,
+    ):
+        """
+        Args:
+            sampler: The sampler used to compute the gradients.
+            epsilon: The offset size for the finite difference gradients.
+            run_options: Backend runtime options used for circuit execution. The order of priority is:
+                run_options in `run` method > gradient's default run_options > primitive's default
+                setting. Higher priority setting overrides lower priority setting.
+        """
+
+        self._epsilon = epsilon
+        super().__init__(sampler, **run_options)
+
+    def _run(
+        self,
+        circuits: Sequence[QuantumCircuit],
+        parameter_values: Sequence[Sequence[float]],
+        parameters: Sequence[Sequence[Parameter] | None] | None = None,
+        **run_options,
+    ) -> SamplerGradientResult:
+        """Compute the sampler gradients on the given circuits."""
+        # if parameters is none, all parameters in each circuit are differentiated.
+        if parameters is None:
+            parameters = [None for _ in range(len(circuits))]
+
+        jobs, result_indices_all = [], []
+        for circuit, parameter_values_, parameters_ in zip(circuits, parameter_values, parameters):
+            # indices of parameters to be differentiated
+            if parameters_ is None:
+                indices = list(range(circuit.num_parameters))
+            else:
+                indices = [circuit.parameters.data.index(p) for p in parameters_]
+            result_indices_all.append(indices)
+
+            offset = np.identity(circuit.num_parameters)[indices, :]
+            plus = parameter_values_ + self._epsilon * offset
+            minus = parameter_values_ - self._epsilon * offset
+            n = 2 * len(indices)
+
+            job = self._sampler.run([circuit] * n, plus.tolist() + minus.tolist(), **run_options)
+            jobs.append(job)
+
+        # combine the results
+        results = [job.result() for job in jobs]
+        gradients, metadata_ = [], []
+        for i, result in enumerate(results):
+            n = len(result.quasi_dists) // 2
+            dists = [Counter() for _ in range(circuits[i].num_parameters)]
+            for j, idx in enumerate(result_indices_all[i]):
+                # plus
+                dists[idx].update(
+                    Counter({k: v / (2 * self._epsilon) for k, v in result.quasi_dists[j].items()})
+                )
+                # minus
+                dists[idx].update(
+                    Counter(
+                        {
+                            k: -1 * v / (2 * self._epsilon)
+                            for k, v in result.quasi_dists[j + n].items()
+                        }
+                    )
+                )
+            gradients.append([QuasiDistribution(dist) for dist in dists])
```

Comment:
Yes, exactly. I was wondering if there are any cooler ways to implement it. It appears in the parameter shift and linear combination gradient as well.

Reply:
It might be simpler to convert QuasiDistribution to ndarray to take the diff of plus and minus.
```python
x = np.zeros(...)
x[list(quasi_dists.keys())] += list(quasi_dists.values())
```